### ISSUE TITLE
Energy totals in VRM portal show up rounded

### ISSUE THREAD

**Issue by adegwert** [schenlap/venus_kostal_pico#5]
Hi, I'm using this great plugin since a while with a PIKO 7.5 (version 1 device with SolarWorld branding). Live values look good and overall regulation with MultiPlus II ESS works fine, too.

The only issue is that the PV energy values in the VRM portal are not precise, for almost each hour they are rounded up to the next kWh.
E. g. the historical data for solar production per-hour shows 1.00, 1.00, 2.00, 3.00, 3.10, 1.00, 1.00 (so not all values are rounded up - cloudy day btw). As a consequence, also the consumption in VRM is miscalculated (as it is the difference between grid meter and PV production).

Thanks upfront for any hints where to look at (or what debug data to provide) :-)

**Comment by adegwert** [comment]
I started to dig into this a little bit - my Kostal is only providing full kWh for the total energy counter. But the daily energy is available with two decimals.
I edited the plugin in kostal.py to read that daily value (EFAT, html line 65 instead of 51) and match those two decimals with the regex expression. Now the values arrive precisely in the VRM portal.

Still testing though, as I am not sure what happens when the daily energy counter from Kostal resets to 0 every day, we will see. Otherwise it might be necessary to store today's total energy in the plugin and add it when the Kostal's counter resets tomorrow.
